openmc/model/surface_composite.py

```diff
@@ -1837,3 +1837,49 @@ def __init__(self, center_base: Sequence[float], axis: Sequence[float],
 
     def __neg__(self) -> openmc.Region:
         return +self.plane_bottom & -self.plane_top & -self.cone
+
+class Z_Vessel(CompositeSurface):
+    """Vessel as a composite surface parallel to z-axis
+        includes composite surfaces with a ZCylinder and 
+        semi-ellipsoids on top and the bottom of this cylinder.
+        """
+
+    _surface_names = ('cycl', 'zmin', 'zmax', 'bottom', 'top')
+
+    def __init__(self, x0, y0, r, zmin, zmax, hbottom, htop, **kwargs):
+        if zmin >= zmax:
+            raise ValueError('zmin must be less than zmax')
+
+        self.cycl = openmc.ZCylinder(x0=x0, y0=y0, r=r, **kwargs)
+        self.zmin = openmc.ZPlane(z0=zmin, **kwargs)
+        self.zmax = openmc.ZPlane(z0=zmax, **kwargs)
+
+        """
+        Coefficients for quadric surface to create an ellipsoid
+
+            General equation for an ellipsoid: 
+                (x-xo)^2/r^2 + (y-yo)^2/r^2 + (z-zo)^2/h^2 = 1
+                
+            General form of a quadric surface equation:
+                Ax^2 + By^2 + Cz^2 + Gx + Hy + Jz + K = 0
+                """
```

Comment:
```suggestion
        """
        General equation for an ellipsoid: 
        (x-xo)^2/r^2 + (y-yo)^2/r^2 + (z-zo)^2/h^2 = 1               
        """
```

Suggesting we reduce this comment to it's critical piece -- we cover the GQ equation in the documentation as it is.